sql/core/src/main/scala/org/apache/spark/sql/DataFrameNaFunctions.scala

```diff
@@ -435,11 +435,10 @@ final class DataFrameNaFunctions private[sql](df: DataFrame) {
    * Returns a [[Column]] expression that replaces null value in `col` with `replacement`.
    */
   private def fillCol[T](col: StructField, replacement: T): Column = {
-    val quotedColName = "`" + col.name + "`"
```

Comment:
Seems like it doesn't support special characters in the column name now. Can we keep the support?

Reply:
I tested special character scenario, it does not affected.
![Screenshot from 2019-08-30 16-54-26](https://user-images.githubusercontent.com/51401130/64017321-f4663900-cb46-11e9-8047-b774a947034e.png)

Is there any other scenario i need to check?

Reply:
Does `.` work too?

```scala
scala> val df = spark.range(1).selectExpr("1 as `a.b`")
df: org.apache.spark.sql.DataFrame = [a.b: int]

scala> df.col("a.b")
org.apache.spark.sql.AnalysisException: Cannot resolve column name "a.b" among (a.b);
  at org.apache.spark.sql.Dataset.$anonfun$resolve$1(Dataset.scala:259)
  at scala.Option.getOrElse(Option.scala:138)
  at org.apache.spark.sql.Dataset.resolve(Dataset.scala:259)
  at org.apache.spark.sql.Dataset.col(Dataset.scala:1340)
  ... 47 elided
```